.github/workflows/notify-merge.js

```diff
@@ -0,0 +1,123 @@
+const fs = require('fs');
+const { slackNotification } = require('./helpers.js');
+
+const SLACK = {
+    merge: ({ html_url, number, title, prefix = '' }) =>
+        `:merged: PR merged to stage: ${prefix} <${html_url}|#${number}: ${title}>.`,
+};
+
+const mergeRegex = /Merge pull request #(\d+)/i;
+function getRepoInfo() {
+    const repoFull = process.env.GITHUB_REPOSITORY;
+    if (!repoFull) {
+        throw new Error("GITHUB_REPOSITORY is not defined in the environment.");
+    }
+    const [owner, repo] = repoFull.split('/');
+    return { owner, repo };
+}
+
+async function getPR(prNumber, owner, repo, token) {
+    const url = `https://api.github.com/repos/${owner}/${repo}/pulls/${prNumber}`;
+    const response = await fetch(url, {
```

Comment:
You could also use `github.rest...` which is a wrapper around Octokit. I have a slight preference towards using that given that this is already handed to us by the action env, is consistent with the rest of the code & in case the github API ever sees changes, octokit would handle that.

Unlikely to see breaking changes in the case of the Github API given it's that widespread, however for consistency sake its still worth it imo